Using different artifact repositories in a restricted environment (#1160)
codenvy/che-docs: src/main/pages/che-7/end-user-guide/proc_defining-maven-repositories-in-the-settings-xml-file.adoc

// Module included in the following assemblies:
//
// using-maven-artifact-repositories

[id="defining-repositories-in-settings-xml_{context}"]
= Defining repositories in `settings.xml`

> You can use the `settings.xml` file, to specify your own artifact repositories at `example.server.org`,

Reviewer: ```suggestion
To specify your own artifact repositories at `example.server.org`, use the `settings.xml` file.
```

> To use your own artifact repositories at `example.server.org`, you must modify the  `settings.xml` file.

Reviewer 2: settings.xml is optional, so "modify" is not the correct term in most cases

> To do that, ensure, that the `settings.xml` is present in all the containers that use Maven tools, in particular the Maven container and the Java plugin container.

Reviewer: ```suggestion
To do that, ensure, that `settings.xml` is present in all the containers that use Maven tools, in particular the Maven container and the Java plug-in container.
```

> By default, `settings.xml` is located at the user home directory, but since user home directory is not on a persistent volume, each time you restart workplace, you must re-create the file.

Reviewer 2: No, it's not: it's located in \<home dir\>/.m2

> By default, `settings.xml` is located at the user home directory, but since user home directory is not on a persistent volume, each time you restart workplace, you must re-create the file.

Reviewer 2: "_because_  _the_ user home directory"

> By default, `settings.xml` is located at the `<home dir>/.m2` directory, but because the user home directory is not on a persistent volume, each time you restart workplace, you must re-create the file. This is a known issue, see link:https://github.com/eclipse/che/issues/13318[the issue on GitHub].

Reviewer: ```suggestion
By default, `settings.xml` is located at the `__<home dir>__/.m2` directory. Because the user home directory is not on a persistent volume, you must re-create the file each time you restart the workspace. This is a known issue, see link:https://github.com/eclipse/che/issues/13318[the issue on GitHub].
```
* Or perhaps just use `~/.m2`?
* Not entirely sure whether my `workplace -> workspace` suggestion is warranted here... Maybe confirm, please?

Reviewer: Ah, based on the next para, workspace is pbbly correct.

> To make the file persist between the restarts of the workspace, create a directory `/projects/maven/.m2` and move the `settings.xml` file in it. Then set the value of the `user.home` Java system property to `/projects/maven` in all your Maven containers.

Reviewer 2: This only makes sense as a workaround for issue https://github.com/eclipse/che/issues/13318. If we don't mention why we're doing this workaround, folks will be confused.

> A workaround is to make the file persist between the restarts of the workspace, create a directory `/projects/maven/.m2` and move the `settings.xml` file in it. Then set the value of the `user.home` Java system property to `/projects/maven` in all your Maven containers.

Reviewer: ```suggestion
A workaround is to make the file persist between the restarts of the workspace, create a directory `/projects/maven/.m2`, and move the `settings.xml` file in it. Then set the value of the `user.home` Java system property to `/projects/maven` in all your Maven containers.
```

             <enabled>true</enabled>
           </releases>
           <snapshots>
             <enabled>false</enabled>
           </snapshots>
           <url>http://example.server.org/repository/maven-releases/</url>
        </pluginRepository>
      </pluginRepositories>
      <repositories>
        <repository>
           <id>my-nexus-snapshots</id>
           <releases>
             <enabled>false</enabled>
           </releases>
           <snapshots>
             <enabled>true</enabled>
           </snapshots>
           <url>http://example.server.org/repository/maven-snapshots/</url>
        </repository>
        <repository>
           <id>my-nexus-releases</id>
           <releases>
             <enabled>true</enabled>
           </releases>
           <snapshots>
             <enabled>false</enabled>
           </snapshots>
           <url>http://example.server.org/repository/maven-releases/</url>
        </repository>
      </repositories>
    </profile>
  </profiles>
  <activeProfiles>
    <activeProfile>my-nexus</activeProfile>
  </activeProfiles>
</settings>
----

. Configure the devfile to use the `/projects/maven` value for the `user.home` Java system property.

> .. For the Java plugin container, configure the language server startup arguments preference:

Reviewer: ```suggestion
.. For the Java plug-in container, configure the language server start arguments preference:
```

+

> ----

Reviewer: ```suggestion
[source,yaml]
----
```

components:
  - id: redhat/java11/latest
    type: chePlugin
    preferences:
      java.jdt.ls.vmargs: >-
        -noverify -Xmx1G -XX:+UseG1GC -XX:+UseStringDeduplication
        -Duser.home=/projects/maven
----

.. For the Maven container, configure the `MAVEN_OPTS` environment variable:
+

> ----

Reviewer: ```suggestion
[source,yaml]
----
```
